src/com/amazon/corretto/crypto/provider/RsaCipher.java

```diff
@@ -177,6 +177,14 @@ protected int engineDoFinal(byte[] input, int inputOffset, int inputLen, final b
                 inputOffset = 0;
                 inputLen = buffer_.size();
             }
+            // One-shot, no input. Cipher only calls engineDoFinal with null input in doFinal overloads
+            // that don't take an input buffer, and in those cases, inputOffset and inputLen are 0.
+            // We set them here anyways to be safe, because the API makes no such guarantee.
+            else if (input == null) {
+                input = new byte[0];
```

Comment:
Please use the zero-byte array constant instead.
```suggestion
                input = Utils.EMPTY_ARRAY;
```